asmcomp/amd64/proc.ml

```diff
@@ -273,15 +273,18 @@ let regs_are_volatile _rs = false
 (* Registers destroyed by operations *)
 
 let destroyed_at_c_call =
+  (* C calling conventions preserve rbp, but it is clobbered
+     by the code sequence used for C calls in emit.mlp, so it
+     is marked as destroyed. *)
   if win64 then
-    (* Win64: rbx, rbp, rsi, rdi, r12-r15, xmm6-xmm15 preserved *)
```

Comment:
Does clobbering rbp have an impact on the performance?

Reply:
I haven't measured, but if this makes performance worse it's a problem in the register allocator.

The current code spills the stack pointer to the C stack around `[@noalloc]` calls, while the new code saves it in `%rbp` instead. If all of the registers are in use, then this change will make the register allocator spill something. Letting the register allocator check whether it needs to spill something seems better than unconditionally spilling `%rsp`, which is definitely in use.